### PR TITLE
CONSOLE-3652: Create StorageVersionMigration for ConsolePlugin CRD v1 migration

### PR DESCRIPTION
**Pull request by jhadvig**
Second part of https://github.com/openshift/api/pull/1477
/assign @JoelSpeed

### CODE REVIEW

File: manifests/0000_90_console-operator_03_console_plugin_storage_migration.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  annotations: 
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+spec:
+  resource:
+    group:  console.openshift.io
+    version: v1
+    resource: consoleplugins
+---
```
Comment (JoelSpeed) [inline manifests/0000_90_console-operator_03_console_plugin_storage_migration.yaml:14]: Nit, probably don't need `---` at the end, and should have a newline char as the final character

**Review by JoelSpeed — COMMENTED** [review]
I think this looks correct, you should be able to test this by creating a cluster, changing the storage version of the CRD to v1, applying this, and then, once processed, should be able to see in the CRD status that stored versions is just `v1`

**Comment by jhadvig** [comment]
@JoelSpeed comment addressed. Will test.

**Comment by jhadvig** [comment]
Tagging the change since we already tested it manually with @JoelSpeed, check for further reasoning https://github.com/openshift/api/pull/1477#issuecomment-1831770288

QE Approver: 
/assign @yufchang 
Docs Approver: 
/assign @opayne1 
PX Approver:
/assign @RickJWagner

**Comment by yanpzhan** [comment]
I will work on the pr as soon as I can.

**Comment by jhadvig** [comment]
ovn contaienr is failing in both e2e-gcp-ovn and e2e-aws-console jobs

**Comment by yanpzhan** [comment]
Checked on cluster launched against the pr, now the version "v1alpha1" for "Storage" is "false", and the version "v1" for "Storage" is "true".
/label qe-approved

**Review by opayne1 — COMMENTED** [review]
/label docs-approved

**Pull request merged**